mlir/lib/Bindings/Python/Presburger.cpp

```diff
@@ -0,0 +1,767 @@
+             return mlirPresburgerIntegerRelationRemoveDuplicateDivs(
+                 self.relation);
+           })
+      .def("simplify",
+           [](PyPresburgerIntegerRelation &self) {
+             return mlirPresburgerIntegerRelationSimplify(self.relation);
+           })
+      .def(
+          "convert_var_kind",
+          [](PyPresburgerIntegerRelation &self,
+             MlirPresburgerVariableKind srcKind, uint64_t vatStart,
+             uint64_t varLimit, MlirPresburgerVariableKind dstKind,
+             py::object &pos) {
+            if (pos.is_none())
+              return mlirPresburgerIntegerRelationConvertVarKindNoPos(
+                  self.relation, srcKind, vatStart, varLimit, dstKind);
+            return mlirPresburgerIntegerRelationConvertVarKind(
+                self.relation, srcKind, vatStart, varLimit, dstKind,
+                pos.cast<uint64_t>());
+          },
+          py::arg("src_kind"), py::arg("start"), py::arg("limit"),
+          py::arg("dst_kind"), py::arg("pos") = py::none())
+      .def(
+          "convert_to_local",
+          [](PyPresburgerIntegerRelation &self,
+             MlirPresburgerVariableKind srcKind, uint64_t vatStart,
+             uint64_t varLimit) {
+            return mlirPresburgerIntegerRelationConvertVarKindNoPos(
+                self.relation, srcKind, vatStart, varLimit,
+                MlirPresburgerVariableKind::Local);
+          },
+          py::arg("src_kind"), py::arg("start"), py::arg("limit"))
+      .def(
+          "add_bound",
+          [](PyPresburgerIntegerRelation &self, MlirPresburgerBoundType type,
+             uint64_t pos, int64_t value) {
+            return mlirPresburgerIntegerRelationAddBound(self.relation, type,
+                                                         pos, value);
+          },
+          py::arg("bound_type"), py::arg("pos"), py::arg("value"))
+      .def(
+          "add_bound",
+          [](PyPresburgerIntegerRelation &self, MlirPresburgerBoundType type,
+             std::vector<int64_t> &expr, int64_t value) {
+            return mlirPresburgerIntegerRelationAddBoundExpr(
+                self.relation, type, expr.data(), expr.size(), value);
+          },
+          py::arg("bound_type"), py::arg("expr"), py::arg("value"))
+      .def(
+          "constant_fold_var",
+          [](PyPresburgerIntegerRelation &self, uint64_t pos) {
+            auto r = mlirPresburgerIntegerRelationConstantFoldVar(self.relation,
+                                                                  pos);
+            return mlirLogicalResultIsSuccess(r);
+          },
+          py::arg("pos"))
+      .def(
+          "constant_fold_var_range",
+          [](PyPresburgerIntegerRelation &self, uint64_t pos, uint64_t num) {
+            return mlirPresburgerIntegerRelationConstantFoldVarRange(
+                self.relation, pos, num);
+          },
+          py::arg("pos"), py::arg("num"))
+      .def_property_readonly(
```

Comment:
@makslevental 
I have added `relation.at_eq/ineq[I, j]` API as you suggested.
However, this is readonly. Looks like we can't set individual element of internal `IntMatrix`.

If this is sufficient, which I think it is, we can remove `equalities`/`inequalities` methods.